include/emp/prefab/QueueManager.hpp

```diff
@@ -0,0 +1,331 @@
+    size_t epochs; // The number of epochs this run is supposed to run for
+
+    RunInfo(SettingConfig _config, size_t _id)
+        : runinfo_config(_config), id(_id), cur_epoch(0) { ; }
+
+    /// @returns current epoch
+    size_t GetEpoch() {return cur_epoch;}
+    /// @returns number of epochs to run for
+    size_t GetNEpochs() {return epochs;}
+    /// Increment current epoch by @param x
+    void IncEpoch(int x = 1) {cur_epoch += x;}
+    /// @returns configuration for this run
+    SettingConfig GetConfig() {return runinfo_config;}
+  };
+
+
+  private:
+  SettingConfig queue_config;
+  std::queue<emp::QueueManager::RunInfo> runs;
+  emp::web::Div display_div;
+  emp::web::TextArea run_input;
+  emp::web::Button queue_button;
+  emp::web::Table display_table;
+
+  emp::vector<std::string> ordered_metric_names;
+  emp::vector<std::function<std::string()>> metric_funs;
+
+  size_t num_runs = 10;
+  bool table_built = false;
+
+  public:
+
+    /// @param user_config An example configuration file for this program. Used to initialize table headers.
+    QueueManager(SettingConfig user_config) : queue_config(user_config) { ; }
+
+    /// @returns True if queue is empty, false if it is not
+    bool IsEmpty() {
+      return runs.empty();
+    }
+
+    /// @returns Number of runs remaining in the queue
+    size_t RunsRemaining() {
+      return runs.size();
+    }
+
+    /// Adds new run to queue using settings specified in @param settings.
+    /// @param epochs indicates how many epochs this run should run for.  
+    void AddRun(SettingConfig settings, size_t epochs) {
+      RunInfo new_run(settings, runs.size());
+      new_run.epochs = epochs;
+      runs.push(new_run);
+    }
+
+    /// Removes run from front of queue
+    void RemoveRun() {
+      emp_assert(!IsEmpty(), "Queue is empty! Cannot remove!");
+      runs.pop();
+    }
+
+    /// @returns The a reference to the first run in the queue
+    /// (i.e. the one that is running currently or, if none are
+    ///  in progress, the next run)
+    RunInfo& FrontRun() {
+      emp_assert(!IsEmpty(), "Queue is empty! Cannot access Front!");
+      return runs.front();
+    }
+
+    /// @returns the Div associated with this QueueManager.
+    emp::web::Div GetDiv() {
+      return display_div;
+    }
+
+    /// Clears the content of the div associated with this QueueManager
+    void ResetDiv() {
+      display_div.Clear();
+      table_built = false;
+    }
+
+    /// Adds table containing information for this QueueManager to the
+    /// Div associated with this QueueManager.
+    ///
+    /// @param id optionally allows you to choose the table's element id,
+    /// for ease of finding it from other parts of your code.
+    ///
+    /// Note that you still need to add this div to your document,
+    /// e.g. `my_doc << my_queue_manager.GetDiv()`;
+    void BuildTable(const std::string & id = "") {
+      emp_assert(!table_built && 
```

Comment:
I'd maybe put the string message as a second parameter to the emp_assert (it'll still get printed out)

Reply:
I kind of prefer the && because then it doesn't get printed out twice, but it's not a strong preference.

Reply:
makes sense